### PR TITLE
Unlock the stream cache on login instead of requiring a separate command

### PR DESCRIPTION
**Pull request by zanderz**
Make Unlock a devel command.
Don't show logged in when unlocked in status command.
(may have more work getting the Electron UI to match)
@maxtaco @patrickxb

### CODE REVIEW

File: go/client/cmd_status.go [hunk 1]
```diff
@@ -199,7 +199,7 @@ func (c *CmdStatus) outputJSON(status *fstatus) error {
 func (c *CmdStatus) outputTerminal(status *fstatus) error {
 	dui := c.G().UI.GetDumbOutputUI()
 	dui.Printf("Username:      %s\n", status.Username)
-	dui.Printf("Logged in:     %s\n", BoolString(status.LoggedInProvisioned, "yes", "no"))
+	dui.Printf("Logged in:     %s\n", BoolString(status.LoggedInProvisioned && status.PassphraseStreamCached, "yes", "no"))
```
Comment (maxtaco) [inline go/client/cmd_status.go]: this might be weird since the JSON answer will show something different. Can you change the field in the struct instead?

**Comment by maxtaco** [comment]
small comment but :+1: otherwise. @patrickxb should sign off too. Thanks Steve!

File: go/engine/login.go [hunk 2]
```diff
@@ -70,6 +70,13 @@ func (e *Login) Run(ctx *Context) error {
 	if err == nil {
 		// login successful
 		e.G().Log.Debug("LoginCurrentDevice.Run() was successful")
+
+		// Unlock the passphrase stream here so the user won't have to
+		// separately
+		if _, err2 := e.G().LoginState().GetPassphraseStream(ctx.SecretUI); err2 != nil {
+			e.G().Log.Debug("Logged in, not handling passphrase stream error %s (err type: %T)", err2, err2)
```
Comment (patrickxb) [inline go/engine/login.go]: Do we want to ignore an error here?  Presumably an error is an invalid passphrase?

It seems if we ignore this error, then the user can be in a state without a passphrase stream cache.

File: go/engine/login.go [hunk 3]
```diff
@@ -70,6 +70,13 @@ func (e *Login) Run(ctx *Context) error {
 	if err == nil {
 		// login successful
 		e.G().Log.Debug("LoginCurrentDevice.Run() was successful")
+
+		// Unlock the passphrase stream here so the user won't have to
```
Comment (patrickxb) [inline go/engine/login.go]: What do you think about moving this code to the LoginCurrentDevice engine?

File: go/client/cmd_status.go (hunk 1 above)
Comment (patrickxb) [inline go/client/cmd_status.go]: Maybe we could just not change the status output so we can see that everything is working in the wild (logged in and passphrase stream cached both always yes).

Comment (maxtaco) [inline go/client/cmd_status.go]: That will confuse users though. If we want to break it out then can we do that further down?

File: go/engine/login.go (hunk 2 above)
Comment (maxtaco) [inline go/engine/login.go]: Yeah you are right.  Bad to swallow the error.

Comment (zanderz) [inline go/engine/login.go]: Wondered about that, thanks guys.

**Comment by zanderz** [comment]
@maxtaco I believe this is what we discussed earlier today? @patrickxb

**Comment by maxtaco** [comment]
Patrick for background Chris Nojima wants the same notion of isloggedin in
the electron app.

Steve I will take a look when I get home!

File: go/libkb/status.go [hunk 4]
```diff
@@ -13,10 +13,11 @@ type UserInfo struct {
...
 
 type CurrentStatus struct {
-	Configured bool
-	Registered bool
-	LoggedIn   bool
-	User       *User
+	Configured      bool
+	Registered      bool
+	LoggedIn        bool
+	LoggedInSession bool
```
Comment (maxtaco) [inline go/libkb/status.go]: can you rename this to `SessionIsValid`?

**Comment by maxtaco** [comment]
one small rename and then go!

**Comment by maxtaco** [comment]
cc: @chrisnojima

**Pull request merged**